### PR TITLE
Overview Example Updates

### PR DESCRIPTION
**Pull request by printerscanner**
# Overview
We're going to get some feedback from the designer next week, but I wanted you all to take a look at this for some feedback first before putting it down until next week. 

## Changes
* Renaming `initial-elements` to `nodes-and-edges`. There is some naming inconsistency across examples. Do we like `nodes-and-edges` for all examples?
* Renaming `TextNode` to `InputNode` to be more descriptive. 
* Deleted `CustomNode` which I believe wasn't needed.
* Made changes to the Resizer input tool, which was not working. 
* Renamed some CSS variables that had ultra-general names to very specific names. I'm curious to hear opinions about what would be the correct convention for these examples. I'm under the impression both that users would copy-paste these examples directly into their code base, so our custom styles should be react-flow specific. Example, changed `edgebutton` to `react-flow__button--edge`, not sure how I feel about it.  
* Additionally, there are a few places with inline styles (that I didn't change) that I feel like would be better to move to classes so there is less that the user has to interpret when looking through the files. Thoughts? Here is an example of a Node that currently has a lot of inline styles:
```
import { memo } from 'react';
import { Handle, Position, NodeResizer } from '@xyflow/react';

function ResizerNode({ data }) {
  return (
    <>
      <NodeResizer minWidth={50} minHeight={50} />
      <Handle type="target" position={Position.Left} />
      <div>{data.label}</div>
      <div
        style={{
          display: 'flex',
          position: 'absolute',
          bottom: 0,
          width: '100%',
          justifyContent: 'space-evenly',
          left: 0,
        }}
      >
        <Handle
          style={{ position: 'relative', left: 0, transform: 'none' }}
          id="a"
          type="source"
          position={Position.Bottom}
        />
        <Handle
          style={{ position: 'relative', left: 0, transform: 'none' }}
          id="b"
          type="source"
          position={Position.Bottom}
        />
      </div>
    </>
  );
}

export default memo(ResizerNode);
```

## xy-theme
Created a theme file called `xy-theme.css`. This contains all of the CSS variables and a few changes to the base classes. Ideally, eventually, these added classes would end up integrated with the library, leaving just the variables. `xy-theme` should eventually live in some parent folder or the library, so all examples can access it. Custom CSS that are specific only to the individual example live in the examples `index.css` file.

One thing about `xy-theme.css`that is probably not that important, but I was getting somewhat existential about is whether we should consider the `node` as a styled entity on its own. Basically, if you style the node like this:

```
.react-flow__node {
  box-shadow: var(--xy-node-boxshadow-default);
  border-radius: var(--xy-node-border-radius-default);
  background-color: var(--xy-node-background-color-default);
  width: 150px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  padding: 10px;
  font-size: 12px;
  flex-direction: column;
  border: var(--xy-node-border-default);
  color: var(--xy-node-color, var(--xy-node-color-default));
}
```
, then whenever you want to have a node that doesn't look like the base node, you'll have to make changes to it. I was finding that this was the preferable method and required a lot less CSS, but I'd be curious to hear if there was a reason why we are currently breaking items out more granularly in the `style.css` base. 

There are a few places in the design where I feel like the design is getting more complicated than the example should be. For example, in the design, we have some custom inputs. If we wanted to implement this for the example, we would need add custom JS. I feel like this can potentially muddy the point of the example.
Screenshot 2024-10-22 at 16 42 39

Another simpler example is this border gradient which is a mildly complicated thing to do with CSS and adds about 50 additional lines to the `index.css` file. Do we consider that kind of example worth it? The point of the example is to show customization, but where should we draw the line when considering user understand-ability?
Screenshot 2024-10-22 at 16 45 37

### CODE REVIEW

**Comment by hayleigh-dot-dev** [comment]
I'll give this a more thorough look tomorrow but on the point of...

> There are a few places in the design where I feel like the design is getting more complicated than the example should be.

I feel like for any example that isn't about _styles_ then this is a non-concern. The huge majority of our examples are about functionality and how they look is not part of the example at all; in all these cases we should feel comfortable doing whatever want with styles.

For the examples where styles is explicitly part of the demonstration we should make sure _those styles_ are easily digestible separate from our theme, but the theme styling should still largely not be a concern for users reading an example. 

---

On the point above I'd move to _hide_ the `xy-theme.css` from the examples viewer (the same way we hide some other files): folks have no reason to see it really.

**Comment by chrtze** [comment]
> * Renaming `initial-elements` to `nodes-and-edges`. There is some naming inconsistency across examples. Do we like `nodes-and-edges` for all examples?

I like initial-elements better because it indicates that this is just the initial data and does not contain any logic or state of the current graph.

> * Renaming `TextNode` to `InputNode` to be more descriptive.

I get your point here but my first thought is that InputNode could be mixed up with our built-in node type `input` (which is just a node with one source handle). Maybe something like `TextInputNode`?

> * Renamed some CSS variables that had ultra-general names to very specific names. I'm curious to hear opinions about what would be the correct convention for these examples. I'm under the impression both that users would copy-paste these examples directly into their code base, so our custom styles should be react-flow specific. Example, changed `edgebutton` to `react-flow__button--edge`, not sure how I feel about it.

I like the idea of making the classnames more specific. However, maybe we can come up with a convention that isn't confused with the built-in library class names? Like `react-flow-example__button-edge`

> * Additionally, there are a few places with inline styles (that I didn't change) that I feel like would be better to move to classes so there is less that the user has to interpret when looking through the files. Thoughts?

Would be good to get rid of the inline styles IMO.

**Comment by peterkogo** [comment]
> I like initial-elements better because it indicates that this is just the initial data and does not contain any logic or state of the current graph.

I think 'elements' is sort of a legacy word in React Flow, right? Anyway, I think I would consider both versions a good naming. 

> [...] changed edgebutton to react-flow__button--edge, not sure how I feel about it.

I think we should  leave the `react/svelte-flow__` prefix to the library and only use normal class names in the examples. We created the prefix to prevent collisions, and if people started adopting the same naming schema it would defeat the point.


> Additionally, there are a few places with inline styles (that I didn't change) that I feel like would be better to move to classes so there is less that the user has to interpret when looking through the files. 

I agree with this. I see the point in using some inline styles in an example, just to show people that this is also possible. But I agree that we should move all styles to classes.
**Or maybe we don't even show people?** I really feel like pumping styles through the node state is terrible design — I think this is only implemented because there are default nodes.


Long story short, I really like the new design! I am so happy we are investing some time into this.

Few nitpicks:
- focus, selected state should have different styles. This was a source of confusion in the past
- maybe also a different hover style? Though, it is cool to see things pop :) 
- Is there a way to make the hover style of the node not trigger when a handle is hovered?
- We should lose the `.overview * { box-sizing: border-box }` because it makes all the handles smaller than they should be
